### ISSUE TITLE
File | Save not saving changes

### ISSUE THREAD

**Issue by rfernand2** [Microsoft/vscode#51689]
Issue Type: <b>Bug</b>

1. make a change to a file (in my case, a python source file)
2. File | Save (or Ctrl-s)
3. open same file in notepad - the change is not there

**A workaround** is to turn off Git support.  Steps for doing that can be found here: [How to turn off git][1].  

**Workaround steps I followed:** File | Preferences | Settings, then pasted the below lines inside the brackets in the user settings panel and then **restarted VS Code.**

    // Is git enabled
    "git.enabled": false,
    
    // Path to the git executable
    "git.path": null,
    
    // Whether auto fetching is enabled.
    "git.autofetch": false,

VS Code version: Code 1.24.0 (6a6e02cef0f2122ee1469765b704faf5d0e0d859, 2018-06-06T17:35:40.560Z)
OS version: Windows_NT x64 10.0.15063

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz (4 x 3392)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.95GB (24.76GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by RMacfarlane** [comment]
Thanks for reporting! I haven't been able to reproduce this on my machine. Does this consistently happen for any projects, or only certain ones? Could you share the rest of your settings?

**Comment by rfernand2** [comment]
Hi Rachel,

It’s a fairly plain install of VS code.  No projects, no add-ins.  Just working on a python file that VS Code happened to notice was associated with git.  I’m happy to send any other info you need – just let me know the specifics.

- Roland

**Comment by RMacfarlane** [comment]
Could you give more details about the file this happens for? Is it part of a git repo and you're only opening VSCode on that file? Does this happen with other files or folders?

**Comment by REEMERGEMEDIA** [comment]
Hi, this is happening to me too and as of yet I have not been able to determine a pattern. I'm working in a React project and it is part of a Git repo.

**Comment by daviseford** [comment]
+1 same issue, plain old react/git repo

**Comment by joaomoreno** [comment]
@rfernand2 Does this reproduce reliably? Does this happen with other file extensions? Is the file within a Git repository? Does this reproduce if the file is not within a Git repository?

**Comment by rfernand2** [comment]
All I know is that I experienced the problem on one day on my home machine.  It kept happening over and over to the same file, which is located in a Git repository on my machine.  As soon as I turned off Git in VS code, the problem disappeared.

On my work machine I have not yet turned off Git in VS Code.  Yesterday, I saw my first problem on that machine with VS Code.  This problem this time as that  I couldn’t type new text in the editor (again it was a *.py file which is mostly all I edit these days).    I restarted VS Code and the problem went away.  I still have Git turned on here, so I’ll let you know if any other problems happen.

-  Roland

**Comment by joaomoreno** [comment]
Let's try to focus on the original issue.

---

Does this reproduce reliably? Does this happen with other file extensions? Is the file within a Git repository? Does this reproduce if the file is not within a Git repository?

**Comment by REEMERGEMEDIA** [comment]
I just experienced this again. (related or not) I disconnected from VPN, then attempted to save. Keyboard shortcuts and menu options for Save did not work. I noticed in the Open Editors pane of Explorer that I had another unsaved file. I selected the file in the pane then tried to save (CMD+S). The file saved as well as did the other file (like I did a Save All). I'll see if I can reproduce, but this happened only once today as I was wrapping things up.
